docs/ref/models/indexes.txt

```diff
@@ -92,3 +92,50 @@ opclasses=['jsonb_path_ops'])`` creates a gin index on ``jsonfield`` using
 ``opclasses`` are ignored for databases besides PostgreSQL.
 
 :attr:`Index.name` is required when using ``opclasses``.
+
+``condition``
+-------------
+
+.. attribute:: Index.condition
+
+.. versionadded:: 2.2
+
+The index can be customized to include a condition. If the table is very large,
+and your queries mostly target a subset, it's convenient to restrict the index
+to that subset.
+
+:attr:`Index.name` is required when using ``condition``.
+
+.. admonition:: Restrictions on PostgreSQL
+
+    PostgreSQL requires functions referenced in the condition to be be marked
+    as IMMUTABLE. Django does not perform any checks on this, and an error will
+    be raised if attempted. This means that functions such as
+    :ref:`date-functions` and :class:`~django.db.models.functions.Concat` are
+    not accepted.
+
+.. admonition:: Restrictions on SQLite
+
+    SQLite imposes different restrictions on how a partial index can be
+    constructed. Please read the documentation
+    <https://www.sqlite.org/partialindex.html> for a complete list.
+
+.. note::
+
+    Oracle does not support partial indexes in this form. Instead, partial
+    indexes are created using a functional index. You should use a
+    :doc:`migration </topics/migrations>` adding the index using SQL instead.
```

Comment:
Maybe:

_"Oracle does not naively support partial indexes. Instead, partial indexes can be emulated using functional indexes. You should use a :doc:`migration </topics/migrations>` adding the index using SQL instead."_

Reply:
Inserted. Thanks :+1: